frontend/src/components/widgets/DataFrame/hooks/useDataEditor.test.ts

```diff
@@ -0,0 +1,331 @@
+/**
```

Comment:
For this test file, I think there should maybe be a TODO for negative test cases. These are all positive test cases but doesn't seem to test whether or not you can delete or add rows when fixedNumRows and etc. I can list out all the negative test cases here but don't think that's super necessary.

Reply:
Good point, I added a few additional test cases to cover those scenarios 👍